moveit_setup_assistant/src/widgets/header_widget.h

```diff
@@ -74,27 +74,50 @@ class LoadPathWidget : public QFrame
   // Stores the path qstring
   QLineEdit* path_box_;
 
+Q_SIGNALS:
+  void pathChanged(const QString& path);
+  void pathEditingFinished();
+
 private Q_SLOTS:
   /// Load the file dialog
   void btn_file_dialog();
 
 public:
   /// Constructor
-  LoadPathWidget(const std::string& title, const std::string& instructions, const bool dir_only = false,
-                 const bool load_only = false, QWidget* parent = 0);
+  LoadPathWidget(const QString& title, const QString& instructions, QWidget* parent, const bool dir_only = false,
+                 const bool load_only = false);
 
   /// Returns the file path in QString format
-  const QString getQPath();
+  QString getQPath() const;
 
   /// Returns the file path in std::string format
-  const std::string getPath();
+  std::string getPath() const;
 
   /// Set the path with QString
   void setPath(const QString& path);
 
   /// Set the path with std string
   void setPath(const std::string& path);
 };
+
+/// extend LoadPathWidget with additional line edit for arguments
+class LoadPathArgsWidget : public LoadPathWidget
```

Comment:
why not just modify the original LoadPathWidget function to support args? this seems overly complicated

Reply:
I had it this way before, but switch to two classes by intention: There are some usages of LoadPathWidget, that do not require the arguments at all. For me, it was clearer to have separate classes for both use cases. It's not more complicated in my opinion.